src/App.tsx

```diff
@@ -9,52 +10,192 @@ import { PostDetails } from './components/PostDetails';
+      throw err;
+    }
+  };
+
+  return (
+    <main className="section">
+      <div className="container">
+        <div className="tile is-ancestor">
+          <div className="tile is-parent">
+            <div className="tile is-child box is-success">
+              <div className="block">
+                {isLoadingUsers || isLoadingPosts ? (
+                  <Loader />
+                ) : (
+                  <UserSelector
+                    selectedUser={selectedUser}
+                    users={users}
+                    onSelect={setSelectedUser}
+                  />
+                )}
               </div>
 
-              <div className="notification is-warning" data-cy="NoPostsYet">
-                No posts yet
+              <div className="block" data-cy="MainContent">
+                {selectedUser && !isLoadingPosts && !error ? (
+                  posts.length > 0 ? (
+                    <PostsList
+                      posts={posts}
+                      selectedPost={selectedPost}
+                      onOpen={setSelectedPost}
+                      onClose={() => {
+                        setSelectedPost(null);
+                      }}
+                    />
+                  ) : (
+                    <div
+                      className="notification is-warning"
+                      data-cy="NoPostsYet"
+                    >
+                      No posts yet
+                    </div>
+                  )
+                ) : (
+                  <p data-cy="NoSelectedUser">No user selected</p>
+                )}
               </div>
 
-              <PostsList />
+              {error && (
+                <div
+                  className="notification is-danger"
+                  data-cy="PostsLoadingError"
+                >
+                  {error}
+                </div>
+              )}
             </div>
           </div>
-        </div>
-
-        <div
-          data-cy="Sidebar"
-          className={classNames(
-            'tile',
-            'is-parent',
-            'is-8-desktop',
-            'Sidebar',
-            'Sidebar--open',
+          {selectedPost && (
+            <div
+              data-cy="Sidebar"
+              className={classNames(
+                'tile',
+                'is-parent',
+                'is-8-desktop',
+                'Sidebar',
+                'Sidebar--open',
+              )}
+            >
+              <div className="tile is-child box is-success ">
+                <PostDetails
+                  post={selectedPost}
+                  error={error}
+                  comments={comments}
+                  formOpen={setIsFormOpen}
+                  isFormOpen={isFormOpen}
+                  deleteComment={handleDeleteComment}
+                  addComment={handleAddComment}
+                  isLoadingComments={isLoadingComments}
+                  commentLoading={() => setIsLoadingComments}
```

Comment:
The `commentLoading` prop is incorrectly set. It should be a function call to update the state, such as `() => setIsLoadingComments(true)` or `() => setIsLoadingComments(false)`, depending on the intended use. Currently, it just passes the function reference `setIsLoadingComments`, which will not work as expected.